modules/services/pyprland.nix

```diff
@@ -0,0 +1,56 @@
+{ options, config, lib, pkgs, ... }:
+let
+  cfg = config.services.pyprland;
+  settingsFormat = pkgs.formats.toml { };
+in with lib; {
+
+  meta.maintainers = [ maintainers.knoc-off ];
+  options.services.pyprland = {
+    enable = mkEnableOption "pyprland";
+
+    package = mkOption {
+      default = pkgs.pyprland;
+      type = types.package;
+      description = "The pyprland package to use";
+    };
+
+    extraPlugins = mkOption {
+      default = [ ];
+      type = with types; listOf str;
+      description = "Additional plugins to enable";
+    };
+
+    settings = mkOption {
+      type = types.submodule {
+        freeformType = settingsFormat.type;
+
+        options.scratchpads = mkOption {
+          default = { };
+          type = with types; attrsOf attrs;
+          description = "Scratchpad configurations";
+        };
+      };
+      default = { };
+      description = ''
+        Configuration for pyprland, see
+        <link xlink:href="https://github.com/hyprland-community/pyprland/wiki/Getting-started#configuring"/>
+        for supported values.
+      '';
+    };
+  };
+
+  config = mkIf cfg.enable {
```

Comment:
Shouldn't this actually start a service? For example,

``` nix
wayland.windowManager.hyprland.settings.exec-once = [(getExe cfg.package)];
```

Reply:
I hadn't thought about this, but yes it makes sense if it does. Ill add the change